account-info/src/lib.rs

```diff
@@ -185,6 +183,22 @@ impl<'a> AccountInfo<'a> {
         Ok(())
     }
 
+    /// Resize the account's data: Either truncating or zero extending.
+    ///
+    /// Note:  Account data can be increased within a single call by up to
+    /// `solana_program::entrypoint::MAX_PERMITTED_DATA_INCREASE` bytes.
+    ///
+    /// # Safety
+    ///
+    /// This method makes assumptions about the layout and location of memory
+    /// referenced by `AccountInfo` fields. It should only be called for
+    /// instances of `AccountInfo` that were created by the runtime and received
+    /// in the `process_instruction` entrypoint of a program.
+    pub fn resize(&self, new_len: usize) -> Result<(), ProgramError> {
```

Comment:
How about marking this `unsafe` in preparation for https://doc.rust-lang.org/nightly/edition-guide/rust-2024/unsafe-op-in-unsafe-fn.html for when `realloc` is removed?

Reply:
The function could be implemented safely, it is just that the way the SDK currently abuses the length field causes it to require an unsafe block inside. Don't see a reason why we should expose that to all call sites.